Libraries/LibGUI/GDirectoryModel.cpp

```diff
@@ -132,6 +134,10 @@ bool GDirectoryModel::fetch_thumbnail_for(const Entry& entry)
         },
 
         [this, path](auto thumbnail) {
+            // this class is disposing, we shouldn't attempt anything.
+            if (m_disposing)
```

Comment:
This is still buggy, since the object the captured `this` points to may have been deleted entirely and so it's not safe to read from `m_disposing`.
Perhaps we could make the destructor of `GDirectoryView` proactively orphan any pending thumbnailing jobs it has scheduled. That way, when they eventually finish, they'll still go into the thumbnail cache, but we'll know not to invoke any callbacks.

Reply:
I'm not sure how to go about doing that.  Would I need to extend BackgroundAction in some way?